### PR TITLE
feat: show vulnerability flags for not affected products

### PR DESCRIPTION
**Pull request by MarcoKorinth**
Fixes https://github.com/secvisogram/secvisogram/issues/505 

# Changes
Show according flags for each product in `Known not affected` list of vulnerability product status.
[Example Test File](https://raw.githubusercontent.com/oasis-tcs/csaf/aa99c2b4475ad113e1de8eb6308edf2bc494eb3d/csaf_2.0/examples/csaf/csaf_vex/sec-vex-2022-0001.json)

image

### CODE REVIEW

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Please also resolve groups and make sure that `label`s of `flags` bound via `group_ids` to products are displayed.
Also, please update the documentation in https://github.com/secvisogram/secvisogram/blob/main/PREVIEW-TEMPLATING.md#full-list-of-extended-template-attributes

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
This fails, if no productGroups are defined...
Moreover: Could you add a test that verifies that the function works correctly?

File: PREVIEW-TEMPLATING.md [hunk 1]
```diff
@@ -465,6 +465,7 @@ This is the full list of document properties that were added to allow the desire
...
 | `vulnerabilities[].product_status.known_not_affected[].vectorString`  | The vectorString of the score in which this product_status is referenced by id.                                                                                                                                                                                 |
 | `vulnerabilities[].product_status.known_not_affected[].baseScore`     | The baseScore of the score in which this product_status is referenced by id.                                                                                                                                                                                    |
+| `vulnerabilities[].product_status.known_not_affected[].flags`         | Comma seperated list of corresponding flags                                                                                                                                                                                                                     |
```
Comment (tschmidtb51) [inline PREVIEW-TEMPLATING.md:468]: For consistency, I guess that should be an array.

**Review by tschmidtb51 — COMMENTED** [review]
I have some minor comment. Let's discuss them in the meeting.

File: PREVIEW-TEMPLATING.md (hunk 1 above)
Comment (MarcoKorinth) [reply PREVIEW-TEMPLATING.md:468]: Yes I agree. I'm going to change it

**Review by tschmidtb51 — APPROVED** [review]
LGTM

**Review by StefanNemeth — APPROVED** [review]

**Pull request merged**